### PR TITLE
Python composite module execution unit tests on delegation of backend_with_compiler_demo

### PR DESCRIPTION
**Pull request by Amyh11325**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60801**

backend_with_compiler_demo

Summary:
Added unit tests for the execution of a simple composite module with a
compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v gives a
success

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29409958](https://our.internmc.facebook.com/intern/diff/D29409958)

### CODE REVIEW

**Comment by Amyh11325** [comment]
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29409958).

File: test/jit/test_backends.py [hunk 1]
```diff
@@ -558,6 +558,66 @@ def test_execution(self):
...
+        lowered_sub = torch._C._jit_to_backend(
+            "backend_with_compiler_demo",
+            torch.jit.script(CompModuleTestWithCompiler.BasicModuleSub()),
+            {"forward": {"": ""}}
+        )
+        self.module = CompModuleTestWithCompiler.CompModule(lowered_add, lowered_sub)
+        self.scripted_module = torch.jit.script(CompModuleTestWithCompiler.CompModule(lowered_add, lowered_sub))
+        # No backend and mobile versions of CompModule currently, so this is filler.
+        self.lowered_module = self.scripted_module
+        self.mobile_module = self.scripted_module
```
Comment (iseeyuan) [inline test/jit/test_backends.py:612]: Would mobile_module work when converting from scripted_module, using _save_to_buffer_for_lite_interpreter(self.scripted_module)?

**Review by iseeyuan — COMMENTED** [review]
Please check the inline comment, and see if it's possible to test mobile module from scripted.

File: test/jit/test_backends.py (hunk 1 above)
Comment (Amyh11325) [reply test/jit/test_backends.py:612]: It seems to work.

**Comment by Amyh11325** [comment]
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29409958).

**Review by iseeyuan — APPROVED** [review]
LGTM. Thanks!

**Pull request closed without merging**